src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1656,6 +1717,16 @@ internal void ErrorReadNotifyUser(string? data)
             }
         }
 
+        /// <summary>Throws a System.ObjectDisposedException if the Proces was disposed</summary>
+        /// <exception cref="System.ObjectDisposedException">If the Proces has been disposed.</exception>
+        private void CheckDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(Process), SR.ProcessDisposed);
```

Comment:
Whatever exception we adopt should be consistent with existing ObjectDisposedExceptions being thrown in the same class, e.g. https://github.com/dotnet/runtime/blob/dd3da2b5e8d4655414f8b6fbe3a97fbe0f8ee8af/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L1153-L1156

Reply:
Used `CheckDisposed` in all places